app/services/courses/content_status_service.rb

```diff
@@ -6,7 +6,7 @@ def execute(enrichment:)
       return :rolled_over if enrichment&.rolled_over?
       return :published if enrichment&.published?
       return :withdrawn if enrichment&.withdrawn?
-      return :published_with_unpublished_changes if enrichment&.has_been_published_before?
+      return :published_with_unpublished_changes if enrichment&.has_been_published_before? || (enrichment&.course&.enrichments&.most_recent && enrichment&.course&.enrichments&.many?)
```

Comment:
why couldn't this be `enrichment.course.published? && enrichment.course.has_unpublished_changes?`?